spec/v1.0-draft/resource_descriptor.md

```diff
@@ -0,0 +1,139 @@
+# ResourceDescriptor field type specification
+
+Version v1.0-draft
+
+A size-efficient description of any mutable or immutable software artifact
+or resource.
+
+## Schema
+
+```json
+{
+  "name": "<NAME>",
+  "uri": "<RESOURCE URI>",
+  "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+  "representation": { /* object */ },
+  "downloadLocation": "<RESOURCE URI>"
+  "mediaType": "<MIME TYPE>",
+  "annotations": { /* object */ }
+}
+```
+
+## Fields
+
+Though all fields are optional, a ResourceDescriptor MUST specify at least
+one of `uri` or `digest` at a minimum.
+
+`name` _string, optional_
+
+> Human-readable identifier to distinguish the resource or artifact locally.
+> The semantics are up to the producer and consumer.
```

Comment:
nit: I think this can be interpreted as two distinct purposes (same problem as Subject.name):

- A human-readable "display name" for debugging, which might not be unique or might change unexpectedly
- A machine-readable "filename" for distinguishing between descriptors, which is expected to be stable so that consumers can rely on it

Can we specify that it's only the latter? In other words, that it is intended to be stable filename or similar so that consumers can rely on it?

Also it's probably worth comparing to `uri`. I interpret this as `name` being local to the attestation, while `uri` is universal.

Reply:
Fair point. Agreed that the `name` should be stable and machine-readable, even if it can also be used as display info for humans to read. Will add these details to the field description.